cmd/server-main.go

```diff
@@ -972,11 +972,37 @@ func serverMain(ctx *cli.Context) {
 	}
 
 	// Initialize users credentials and policies in background right after config has initialized.
+	globalIAMFullyInitialized.Store(true)
 	go func() {
 		bootstrapTrace("globalIAMSys.Init", func() {
 			globalIAMSys.Init(GlobalContext, newObject, globalEtcdClient, globalRefreshIAMInterval)
 		})
 
+		// Start a thread to keep checking IAM initialized and re-trigger
+		go func() {
+			ticker := time.NewTicker(10 * time.Second) // Check every 10 seconds
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					if !globalIAMFullyInitialized.Load() {
+						globalIAMSys.Init(GlobalContext, newObject, globalEtcdClient, globalRefreshIAMInterval)
+						if globalIAMFullyInitialized.Load() && globalBrowserEnabled {
+							bootstrapTrace("initConsoleServer", func() {
```

Comment:
Isn't this called twice? We also call it below - and it never waits for IAM, since everything is in a goroutine.

Reply:
I felt outer go routine would have exited by this time if all success and the inner one is running a ticker and first Init failed while start of server, it would keep waiting till it succeeds and then we set the console configuration afresh. Without this even if Init was successful console was not updated of openid availability and was not throwing option to login using openid.
Not sure if this understanding is wrong though.